sdk/search/search-documents/samples/javascript/src/dataSourceConnections/createDataSourceConnection.js

```diff
@@ -13,6 +13,11 @@ const apiKey = process.env.SEARCH_API_KEY || "";
 async function main() {
   console.log(`Running Create Datasource Connection Sample....`);
 
+  if (!endpoint || !apiKey) {
```

Comment:
would it make more sense to put this logic into the Client/Credential constructors? That is to say, have SearchIndexerClient/SearchClient/etc throw if the endpoint is the empty string (falsy) and same for AzureKeyCredential?

Reply:
I will create a separate issue to look into this. Any case, we do not want the error to be thrown in the samples to make the CI pass